toolchain/base/value_store.h

```diff
@@ -125,12 +182,18 @@ class ValueStore
   auto Reserve(size_t size) -> void { values_.reserve(size); }
 
   // These are to support printable structures, and are not guaranteed.
-  auto Print(llvm::raw_ostream& out) const -> void { Print(out, 0); }
-  auto Print(llvm::raw_ostream& out, int indent) const -> void {
-    for (const auto& value : values_) {
-      out.indent(indent);
-      out << "- " << value << "\n";
-    }
+  auto Print(llvm::raw_ostream& out) const -> void {
+    Print(out, std::nullopt, 0, 0);
+  }
+  auto Print(llvm::raw_ostream& out, std::optional<llvm::StringRef> label,
+             int first_line_indent, int later_indent,
+             // This decays so that `const llvm::APInt` printing catches the
+             // specialization.
+             PrintFn print = Internal::PrintValue<std::decay_t<ValueT>>) const
+      -> void {
+    PrintValueRange(out, llvm::iterator_range(values_), label,
+                    first_line_indent, later_indent, /*trailing_newline=*/true,
+                    print);
   }
```

Comment:
Is it worth factoring this out to a CRTP base class? We seem to do the same thing three times so far, and I expect there'll be more as we add more value stores. I suppose we'll need to add an accessor to get the value range, but given that you're only supporting an `iterator_range<const ValueT*>` right now, it seems like we could use the `array_ref` accessor for now at least.